pkg/generate/generate.go

```diff
@@ -43,18 +45,28 @@ func Generate(fh *hcl2nix.FileHandlers, sc buildsafev1.SearchServiceClient) erro
 	}
 
 	lang := findLang(conf)
-	if lang == "" {
-		return fmt.Errorf("could not detect programming language of the app")
-	}
 
 	cr := hcl2nix.ResolveCategoryRevisions(conf.Packages, lockPackages)
-	err = btemplate.GenerateFlake(btemplate.Flake{
-		// Description:         "bsf flake",
+	genFlake := btemplate.Flake{
 		NixPackageRevisions: cr.Revisions,
-		DevPackages:         cr.Development,
-		RuntimePackages:     cr.Runtime,
 		Language:            string(lang),
-	}, fh.FlakeFile, conf)
+	}
+
+	switch pkgType {
+	case "dev":
+		genFlake.DevPackages = cr.Development
+	case "runtime":
+		genFlake.RuntimePackages = cr.Runtime
+	case "all":
+		genFlake.DevPackages = cr.Development
+		genFlake.RuntimePackages = cr.Runtime
+	}
+
+	if devDeps {
+		conf.OCIArtifact[0].DevDeps = true
```

Comment:
this doesn't look right..we're picking the first OCIArtifact, it may not be `pkgs` but this function cannot know ahead of time the OCI attribute user wants to build 🤔

One thing we could do is modify the template to have both dev and runtime attributes. So the `Generate` function doesn't have to care about this and when a user does `bsf oci `(with or without --dev) , we'll already have the Nix side generated.

Reply:
fixed